kernel/arch/dreamcast/kernel/timer.c

```diff
@@ -212,16 +222,24 @@ void timer_ms_disable(void) {
     timer_disable_ints(TMU2);
 }
 
-static void timer_getticks(uint32_t *secs, uint32_t *ticks, uint32_t div) {
+/* Generic function for retrieving the current time maintained by TMU2. 
+   Returns the total amount of time that has elapsed since KOS has been
+   initialized.
+
+   "secs" are directly returned based on the current value of the second
+   counter, with care taken to address pending underflow.
+
+   "ticks" are the subsecond component which are calculated by directly 
+   using TMU2's current counter value. Their resolution is given by "res,"
+   which is the number of nanoseconds within a single tick. */
+static void timer_getticks(uint32_t *secs, uint32_t *ticks, uint32_t res) {
     const int irq_status = irq_disable();
 
     if(secs) {
-        *secs = timer_ms_sec_counter;
-
-        /* Overflow is only notable if we have seconds we can
-           overflow into, so avoid read of TCR if secs is null */
-        if(TIMER16(tcrs[TMU2] & (1 << UNF)))
-            (*secs)++;
+        /* Underflow is only notable if we have seconds we can
```

Comment:
Shouldn't this comment be on the outside of the if block?

Reply:
Oh snap. YES.